cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/TetherControlMessage.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tether;
+
+import com.google.gson.JsonElement;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Control messages sent from tether server to the client.
+ */
+public class TetherControlMessage {
+  public enum Type {
+    KEEPALIVE,
+    RUN_PIPELINE
+  }
+  private Type type;
+  @Nullable
+  private JsonElement payload;
```

Comment:
why do we need it to be json? Can it be bytes?

Reply:
changed to bytes[]